### PR TITLE
cw20-ics20 'Amount' address validation and tests

### PR DESCRIPTION
**Pull request by M-Daeva**
cw20-ics20
1. Added address verification for Amount::Cw20
2. Added tests for Amount::from_parts

### CODE REVIEW

**Comment by uint** [comment]
Hi! Thank you for the submission.

How is this `LocalApi` different from `cosmwasm_std::testing::MockApi`? At a glance it seems to do the exact same thing.

**Comment by uint** [comment]
I also don't think the logic we use in our mocks is supposed to be used in a production setting. I suspect this won't validate an address properly at all.

File: contracts/cw20-ics20/src/amount.rs [hunk 1]
```diff
@@ -1,30 +1,32 @@
...
-use cw20::Cw20Coin;
+use cosmwasm_std::{Addr, CanonicalAddr, Coin, StdError, StdResult, Uint128};
+use cw20::Cw20CoinVerified;
 use std::convert::TryInto;
 
 #[cw_serde]
 pub enum Amount {
     Native(Coin),
-    // FIXME? USe Cw20CoinVerified, and validate cw20 addresses
-    Cw20(Cw20Coin),
```
Comment (ethanfrey) [inline contracts/cw20-ics20/src/amount.rs:11]: I wouldn't do this. You can see the problem with trying to parse a string and suddenly needing `Deps` to do the verification.

I would rather keep Amount as is, and then make an `VerifiedAmount` type, with

```rust
impl Amount {
  pub fn verify(self, deps: Deps) -> StdResult<VerifiedAmount> { /* ... */ }
}
```

Moving parsing and verifying into two steps.

File: contracts/cw20-ics20/src/amount.rs [hunk 2]
```diff
@@ -64,3 +66,158 @@ impl Amount {
...
+            Amount::from_parts(cw20_denom.into(), amount),
+            Amount::Cw20(Cw20CoinVerified {
+                address: cw20_addr,
+                amount
+            })
+        );
+    }
+
+    #[test]
+    #[should_panic]
```
Comment (ethanfrey) [inline contracts/cw20-ics20/src/amount.rs:216]: Definitely panic is also bad, which is why changing API and returning Result on a verify method is cleaner

**Comment by uint** [comment]
Closing due to inactivity

**Pull request closed without merging**